src/azure-cli/azure/cli/command_modules/vm/_image_builder.py

```diff
@@ -455,13 +462,86 @@ def create_image_template(  # pylint: disable=too-many-locals, too-many-branches
         else:
             logger.info("No applicable destination found for destination %s", str(tuple([dest_type, rid, loc_info])))
 
+    # Identity
+    identity_body = None
+    if identity is not None:
+        subscription_id = get_subscription_id(cmd.cli_ctx)
+        user_assigned_identities = {}
+        for ide in identity:
+            if not is_valid_resource_id(ide):
+                ide = resource_id(subscription=subscription_id, resource_group=resource_group_name,
+                                  namespace='Microsoft.ManagedIdentity', type='userAssignedIdentities', name=ide)
+            user_assigned_identities[ide] = ImageTemplateIdentityUserAssignedIdentitiesValue()
+        identity_body = ImageTemplateIdentity(type='UserAssigned', user_assigned_identities=user_assigned_identities)
+
     image_template = ImageTemplate(source=template_source, customize=template_scripts, distribute=template_destinations,
-                                   location=location, build_timeout_in_minutes=build_timeout, tags=(tags or {}))
+                                   location=location, build_timeout_in_minutes=build_timeout, tags=(tags or {}),
+                                   identity=identity_body)
 
     return cached_put(cmd, client.virtual_machine_image_templates.create_or_update, parameters=image_template,
                       resource_group_name=resource_group_name, image_template_name=image_template_name)
 
 
+def _underscore_camel(name):
+    """
+    Transform underscore style to camel style
+    :param name:
+    :return:
+    """
+    terms = name.split('_')
+    if len(terms) == 1:
+        return name
+    return terms[0] + ''.join(map(lambda x: x.title(), terms[1:]))
+
+
+def _serialize(body):
```

Comment:
In my mind, serialization is the process to translate a object to a string. It seems it's not a normal serialization, maybe `transfer` or `convert` is more accurate.

Reply:
You are right. The return value is not a string.
It's deleted now.